prometheus/histogram.go

```diff
@@ -319,9 +319,9 @@ func (m *HistogramVec) GetMetricWithLabelValues(lvs ...string) (Histogram, error
 }
 
 // GetMetricWith replaces the method of the same name in MetricVec. The
-// difference is that this method returns a Histogram and not a Metric so that no
+// difference is that this method returns a Observer and not a Metric so that no
 // type conversion is required.
-func (m *HistogramVec) GetMetricWith(labels Labels) (Histogram, error) {
+func (m *HistogramVec) GetMetricWith(labels Labels) (Observer, error) {
 	metric, err := m.MetricVec.GetMetricWith(labels)
 	if metric != nil {
 		return metric.(Histogram), err
```

Comment:
I'm 100% cool with this, and was personally wondering when I wrote this if that's something we should do. The code still compiles and executes fine, so I'm wondering if there's some sort of implicit conversion on return? I'm also wondering what effect/benefit this has, since as mentioned, the code does still compile and "run fine".